apps/aecore/lib/aecore/multi_node_test_framework/worker.ex

```diff
@@ -0,0 +1,619 @@
+    )
+
+    send_command(node_name, "Pool.add_transaction(spend)")
+    mine_sync_block(node_name)
+
+    send_command(
+      node_name,
+      "next_nonce = Account.nonce(Chain.chain_state().accounts, transfer_to_pub) + 1"
+    )
+
+    send_command(
+      node_name,
+      "{:ok, revoke} = Account.name_revoke(transfer_to_pub, transfer_to_priv, \"test.aet\", 10, next_nonce)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(revoke)")
+  end
+
+  @spec chainstate_naming(String.t()) :: :ok | :unknown_node
+  def chainstate_naming(node_name) do
+    send_command(node_name, "Chain.chain_state().naming")
+  end
+
+  @doc """
+    Gets all peers for a given node
+  """
+  @spec get_all_peers(String.t()) :: :ok
+  def get_all_peers(node_name) do
+    send_command(node_name, "peers = Peers.all_peers")
+
+    send_command(
+      node_name,
+      "peers_encoded = Enum.reduce(peers, [], fn (peer), acc ->
+                                acc ++ [put_in(peer.pubkey, Base.encode32(peer.pubkey))]
+                              end)"
+    )
+
+    # encoding the peers map to the json
+    send_command(node_name, "{:ok, json} = Poison.encode(peers_encoded)")
+
+    # writing the json to the file
+    send_command(node_name, "path = System.cwd() <> \"/result.json\"")
+    send_command(node_name, "File.write(path, json)")
+
+    # updating peers map in the state after 1 second
+    :timer.sleep(1000)
+    update_peers_map(node_name)
+  end
+
+  def busy_port?(port) do
+    :os.cmd('lsof -i -P -n | grep -w #{port}') != []
+  end
+
+  # server
+
+  def handle_info({_, {:data, result}}, state) do
+    if String.match?(result, ~r/error/) do
+      Logger.error(fn -> result end)
+    end
+
+    {:noreply, state}
+  end
+
+  def handle_call({:update_oracle_interaction_objects_state, node_name}, _, state) do
+    path = state[node_name].path <> "/result.json"
+
+    # decoding all the keys and the data to it's initial state
+    with {:ok, data} <- File.read(path),
+         {:ok, decoded_data} <- Poison.decode(data) do
+      oracles_decode32 = for {k, v} <- decoded_data, into: %{}, do: {Base.decode32!(k), v}
+
+      decoded_data =
+        Enum.reduce(oracles_decode32, %{}, fn {k, val}, _ ->
+          atom_keys_map = for {nested_k, v} <- val, into: %{}, do: {String.to_atom(nested_k), v}
+          new_map = put_in(oracles_decode32[k], atom_keys_map)
+          new_map = put_in(new_map[k].oracle_address, Base.decode32!(new_map[k].oracle_address))
+          put_in(new_map[k].sender_address, Base.decode32!(new_map[k].sender_address))
+        end)
+
+      # updating the state
+      new_state = put_in(state[node_name].oracle_interaction_objects, decoded_data)
+
+      # deleting the file result.json
+      File.rm(path)
+      {:reply, :ok, new_state}
+    else
+      {:error, reason} -> {:reply, reason, state}
+    end
+  end
+
+  def handle_call({:update_registered_oracles_state, node_name}, _, state) do
```

Comment:
Because this handle_call is quite similar to `:update_oracle_interaction_objects_state`, what you could do is have a common function with the `with statement` and to put the different part in an anonymous function and pass it to the common function and execute it inside.

You could make this change in couple of places, I don't think I need to point you all of them.

Example:
```elixir

def handle_call() do
  function = fn() -> 
        Enum.reduce(oracles_decode32, %{}, fn {k, val}, _ ->
          atom_keys_map = for {nested_k, v} <- val, into: %{}, do: {String.to_atom(nested_k), v}
          new_map = put_in(oracles_decode32[k], atom_keys_map)
          new_map = put_in(new_map[k].oracle_address, Base.decode32!(new_map[k].oracle_address))
          put_in(new_map[k].sender_address, Base.decode32!(new_map[k].sender_address))
        end)  
   end
   common_func(function)
end

def common_func(function) do
    with {:ok, data} <- File.read(path),
         {:ok, decoded_data} <- Poison.decode(data) do
      oracles_decode32 = for {k, v} <- decoded_data, into: %{}, do: {Base.decode32!(k), v}
       function.(oracle_decode32)

         # deleting the file result.json
       File.rm(path)
       {:reply, :ok, new_state}
     else
       {:error, reason} -> {:reply, reason, state}
     end
   end
end  
```
This is a rough example, but I think you got my point.